src/flamegraph/color/mod.rs

```diff
@@ -16,49 +16,88 @@ const BLUE_GRADIENT: (&str, &str) = ("#eeeeee", "#e0e0ff");
 const GREEN_GRADIENT: (&str, &str) = ("#eef2ee", "#e0ffe0");
 const GRAY_GRADIENT: (&str, &str) = ("#f8f8f8", "#e8e8e8");
 
+/// A flame graph background color.
+///
+/// Defaults to yellow.
 #[derive(Clone, Copy, Debug)]
 pub enum BackgroundColor {
+    /// A yellow gradient from `#EEEEEE` to `#EEEEB0`.
     Yellow,
+    /// A blue gradient from `#EEEEEE` to `#E0E0FF`.
     Blue,
+    /// A green gradient from `#EEF2EE` to `#E0FFE0`.
     Green,
+    /// A grey gradient from `#F8F8F8` to `#E8E8E8`.
     Grey,
+    /// A flag background color with the given RGB components.
+    ///
+    /// Expressed in string form as `#RRGGBB` where each component is written in hexadecimal.
     Flat(u8, u8, u8),
 }
 
+impl Default for BackgroundColor {
+    fn default() -> Self {
+        BackgroundColor::Yellow
+    }
+}
+
+/// A flame graph color palette.
+///
+/// Defaults to [`BasicPalette::Hot`].
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum Palette {
+    /// A plain color palette in which the color is independent of the function name.
     Basic(BasicPalette),
+    /// A semantic color palette in which different hues are used to signifiy semantic aspects of
+    /// different function names (kernel functions, JIT functions, etc.).
     Multi(MultiPalette),
 }
 
+impl Default for Palette {
+    fn default() -> Self {
+        Palette::Basic(BasicPalette::Hot)
+    }
+}
+
+/// A plain color palette in which the color is independent of the function name.
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum BasicPalette {
+    /// A palette in which colors are chosen randomly from a red-yellow spectrum.
     Hot,
+    /// A palette in which colors are chosen randomly from a green-blue spectrum.
     Mem,
+    /// A palette in which colors are chosen randomly from a wide blue spectrum.
     Io,
+    /// A palette in which colors are chosen randomly from a red spectrum.
     Red,
+    /// A palette in which colors are chosen randomly from a green spectrum.
     Green,
+    /// A palette in which colors are chosen randomly from a blue spectrum.
     Blue,
+    /// A palette in which colors are chosen randomly from an aqua-tinted spectrum.
     Aqua,
+    /// A palette in which colors are chosen randomly from a yellow spectrum.
     Yellow,
+    /// A palette in which colors are chosen randomly from a purple spectrum.
     Purple,
+    /// A palette in which colors are chosen randomly from a orange spectrum.
     Orange,
 }
 
+/// A semantic color palette in which different hues are used to signifiy semantic aspects of
+/// different function names (kernel functions, JIT functions, etc.).
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum MultiPalette {
+    /// Use Java semantics to color frames.
     Java,
+    /// Use JavaScript semantics to color frames.
     Js,
+    /// Use Perl semantics to color frames.
     Perl,
+    /// TODO: this seems to just be [`BasicPalette::Aqua`]?
```

Comment:
The difference between `BasicPalette::Aqua` and `MultiPalette::Wakeup` is the default background gradient (`Grey` for `Aqua`, `Blue` for `Wakeup`).